flake.nix

```diff
@@ -14,65 +12,96 @@
+        nixpkgs.lib.genAttrs (import systems) (
+          system:
+          f (
+            import nixpkgs {
+              inherit system;
+              config.allowUnfree = true;
+            }
+          )
+        );
     in
     {
-      packages = forEachSystem (system: {
-        devenv-up = self.devShells.${system}.default.config.procfileScript;
-        devenv-test = self.devShells.${system}.default.config.test;
-      });
-
       nixosConfigurations.tikpannu = nixpkgs.lib.nixosSystem {
         specialArgs = { inherit inputs; };
         modules = [ ./tikpannu-nixos-config/configuration.nix ];
       };
 
-      devShells = forEachSystem (
-        system:
-        let
-          pkgs = import nixpkgs {
-            inherit system;
-            config.allowUnfree = true;
-          };
-        in
+      checks = forAllSystems (
+        pkgs:
         {
-          default = devenv.lib.mkShell {
-            inherit inputs pkgs;
-            modules = [
+          formatting =
+            pkgs.runCommand "fmt-check"
               {
-                # https://devenv.sh/reference/options/
-                packages = with pkgs; [
-                  (azure-cli.withExtensions [
-                    azure-cli-extensions.ssh
-                  ])
-                  sops
-                ];
-                languages.terraform.enable = true;
-
-                git-hooks.hooks.terraform-fmt = {
-                  enable = true;
-                  name = "Terraform fmt check";
-                  entry = "terraform fmt --recursive";
-                  pass_filenames = false;
-                };
+                nativeBuildInputs = [ self.formatter.${pkgs.stdenv.hostPlatform.system} ];
               }
-            ];
+              ''
+                cp -r ${self} repo
+                chmod -R +w repo/
+                treefmt --ci --tree-root repo
+                touch $out
+              '';
+        }
+        // (import ./tikpannu-nixos-config/tests { inherit pkgs inputs; })
+      );
+
+      formatter = forAllSystems (
+        pkgs:
+        pkgs.treefmt.withConfig {
+          runtimeInputs = with pkgs; [
+            nixfmt
+            opentofu
+            yamlfmt
+          ];
+
+          settings.formatter = {
+            nix = {
+              command = "nixfmt";
+              includes = [ "*.nix" ];
+            };
+            terraform = {
+              command = "tofu";
+              options = [ "fmt" ];
+              includes = [ "*.tf" ];
+            };
+            yaml = {
+              command = "yamlfmt";
+              includes = [
+                "*.yaml"
+                "*.yml"
+              ];
+            };
           };
         }
       );
+
+      devShells = forAllSystems (pkgs: {
+        default = pkgs.mkShellNoCC {
+          packages = with pkgs; [
+            (azure-cli.withExtensions [
+              azure-cli-extensions.ssh
+            ])
+            sops
+            opentofu
```

Comment:
I don't think this is supported, as the registries we currently have in use don't work with `tofu`, i.e. commands like `tofu init` and `tofu plan` will not work. Did you try this out?